pkg/segment/query/metricsquery.go

```diff
@@ -116,13 +123,30 @@ func ApplyMetricsQuery(mQuery *structs.MetricsQuery, timeRange *dtu.MetricsTimeR
 		return mRes
 	}
 
-	errors = mRes.ApplyFunctionsToResults(parallelism, mQuery.Function)
-	if errors != nil {
-		for _, err := range errors {
-			mRes.AddError(err)
+	for mQuery.MQueryAggs != nil {
+		if mQuery.MQueryAggs.AggBlockType == structs.FunctionBlock {
+			mQuery.Function = *mQuery.MQueryAggs.FunctionBlock
+			errors = mRes.ApplyFunctionsToResults(parallelism, mQuery.Function)
+			if errors != nil {
+				for _, err := range errors {
+					mRes.AddError(err)
+				}
+
+				return mRes
+			}
+		} else if mQuery.MQueryAggs.AggBlockType == structs.AggregatorBlock {
+			mQuery.Aggregator = *mQuery.MQueryAggs.AggregatorBlock
+			errors = mRes.ApplyAggregationToResults(parallelism, mQuery.Aggregator)
+			if errors != nil {
+				for _, err := range errors {
+					mRes.AddError(err)
+				}
+
+				return mRes
+			}
 		}
```

Comment:
nit: add an `else` block to log an error if we get an invalid `mQuery.MQueryAggs.AggBlockType`